src/schema/v2/artist/targetSupply/index.ts

```diff
@@ -0,0 +1,124 @@
+import { ResolverContext } from "types/graphql"
+import { getMicrofunnelData } from "./utils/getMicrofunnelData"
+import { shuffle } from "lodash"
+
+import {
+  GraphQLObjectType,
+  GraphQLBoolean,
+  GraphQLFieldConfig,
+  GraphQLList,
+  GraphQLString,
+} from "graphql"
+import { ArtworkType } from "schema/v2/artwork"
+
+const ArtistTargetSupplyType = new GraphQLObjectType<any, ResolverContext>({
+  name: "ArtistTargetSupply",
+  fields: {
+    isInMicrofunnel: {
+      description:
+        "Returns whether an artist is in within the microfunnel list.",
+      type: GraphQLBoolean,
+      resolve: artist => Boolean(getMicrofunnelData(`/artist/${artist.id}`)),
+    },
+
+    microfunnel: {
+      type: new GraphQLObjectType<any, ResolverContext>({
+        name: "ArtistTargetSupplyMicrofunnel",
+        fields: {
+          /**
+           * This field is resolved by parsing static CSVtoJSON data.
+           * @see src/schema/v2/artist/targetSupply/utils/getMicrofunnelData.ts
+           */
+          metadata: {
+            type: new GraphQLObjectType<any, ResolverContext>({
+              name: "ArtistTargetSupplyMicrofunnelMetadata",
+              fields: {
+                highestRealized: {
+                  type: GraphQLString,
+                },
+                realized: {
+                  type: GraphQLString,
+                },
+                recentlySoldArtworkIDs: {
+                  type: new GraphQLList(GraphQLString),
+                },
+                roundedUniqueVisitors: {
+                  type: GraphQLString,
+                },
+                roundedViews: {
+                  type: GraphQLString,
+                },
+                str: {
+                  type: GraphQLString,
+                },
+                uniqueVisitors: {
+                  type: GraphQLString,
+                },
+                views: {
+                  type: GraphQLString,
+                },
+              },
+            }),
+          },
+          /**
+           * Take all of the recentlySoldArtworkIDs from metadata and perform
+           * a fetch for associated artworks and attach recently sold prices.
+           */
+          artworks: {
+            args: {
+              randomize: {
+                type: GraphQLBoolean,
+                description:
+                  "Randomize the order of artworks for display purposes.",
+              },
+            },
+            type: new GraphQLList(
+              new GraphQLObjectType({
+                name: "ArtistTargetSupplyMicrofunnelArtwork",
+                fields: () => {
```

Comment:
Note how we're using a function here. As this type is referring to `ArtworkType`, and somewhere in the Artwork type theres a file that's referenced in the Artist type, it introduced a circular dependency somewhere. By using the lazy function form of `fields` we can work around it.

Reply:
who woulda 'thunk' it!